.github/workflows/build-yocto.yml

```
@@ -13,7 +13,25 @@ env:
   BASE_ARTIFACT_URL: "https://quic-yocto-fileserver-1029608027416.us-central1.run.app/${{ github.run_id }}"
 
 jobs:
+  kas-mirror:
+    if: github.repository == 'qualcomm-linux/meta-qcom'
+    runs-on: [self-hosted, x86]
+    steps:
+      - name: Update kas mirrors
+        run: |
+          for r in $(find ${KAS_REPO_REF_DIR}/* -maxdepth 0 -type d); do
+            echo "pre-fetch: $r"
+            for i in $(seq 3); do
+              if git -C $r fetch --prune origin '+refs/*:refs/*'; then
```

if it fails 3 times, the for loop will still exit with a success exit code, and the job won't fail, no? I think it's ok to run git just once and let the job fail, if it ever happens.

---

Nice catch! yeah I can remove the loop and keep it simple.  When the git fails we can manually relaunch only that job.